File: fiyaback/src/main/java/com/app/fiya/field/dto/FieldDetailResponse.java
package com.app.fiya.field.dto;

import com.app.fiya.appointment.dto.AppointmentFieldResponse;
import com.app.fiya.field.model.Field;
import com.app.fiya.rent.dto.RentResponse;
import com.app.fiya.rent.model.Rent;
import com.app.fiya.user.dto.UserTeamResponse;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.Set;
import java.util.stream.Collectors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class FieldDetailResponse {
    private Long id;
    private String name;
    private String latitude;
    private String longitude;
    private double price;
    private int teamCapacity;
    private String ground;
    private String description;
    private Set<AppointmentFieldResponse> appointments;
    //private Set<RentResponse> renting;

    public static FieldDetailResponse of (Field data){
        return FieldDetailResponse.builder()
                .id(data.getId())
                .name(data.getName())
                .latitude(data.getLatitude())
                .longitude(data.getLongitude())
                .price(data.getPrice())
                .teamCapacity(data.getTeamCapacity())
                .ground(data.getGround().toString())
                .description(data.getDescription())
                .appointments(data.getAppointments().stream()
                        .map(AppointmentFieldResponse::fromAppointment)
                        .collect(Collectors.toSet()))
                .build();
    }
}

File: fiya_front/lib/models/field/field_detail_response.dart
class FieldDetailResponse {
  int? id;
  String? name;
  String? latitude;
  String? longitude;
  double? price;
  int? teamCapacity;
  String? ground;
  String? description;

  FieldDetailResponse(
      {this.id,
      this.name,
      this.latitude,
      this.longitude,
      this.price,
      this.teamCapacity,
      this.ground,
      this.description});

  FieldDetailResponse.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    name = json['name'];
    latitude = json['latitude'];
    longitude = json['longitude'];
    price = json['price'];
    teamCapacity = json['teamCapacity'];
    ground = json['ground'];
    description = json['description'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['name'] = this.name;
    data['latitude'] = this.latitude;
    data['longitude'] = this.longitude;
    data['price'] = this.price;
    data['teamCapacity'] = this.teamCapacity;
    data['ground'] = this.ground;
    data['description'] = this.description;
    return data;
  }
}
